tests/s25Main/mapGenerator/testAlgorithms.cpp

```diff
@@ -24,6 +24,36 @@ using namespace rttr::mapGenerator;
 
 BOOST_AUTO_TEST_SUITE(AlgorithmsTests)
 
+BOOST_AUTO_TEST_CASE(join_vector_of_simple_vectors)
+{
+    std::vector<std::vector<int>> multiVector{{1, 2, 3}, {2, 3, 4}, {5, 6}};
+    std::vector<int> result = join(multiVector);
```

Comment:
You can directly do `BOOST_TEST(join(multiVector) == std::vector<int>{1, 2, 3, 2, 3, 4, 5, 6}, boost::test_tools::per_element())` (Check others tests, might misremember exact usage)
Or keep the `result` (const of course) to also check that the return type is as expected